### ISSUE TITLE
Apply low-pass filter when down-sampling to avoid aliasing

### ISSUE THREAD

**Issue by angerhang** [OxWearables/biobankAccelerometerAnalysis#155]

**Comment by chanshing** [comment]
Just to add a bit more clarification: As Dan said, the current resampling implementation is really not intended for resampling to arbitrary rates, but just to correct the configured sampling rate of the device (so if configured at 100Hz, use 100Hz). The resampling is doing simple linear/nearest interpolation, which is not great when say dowsampling by a factor or 2 and beyond without antialiasing filters. Because we ask users to supply the sample rate, it is misleading because gives the impression that we support arbitrary downsampling/upsampling. Looking at the code, I think there's nothing preventing us to just grab the configured sampling rate from the device instead of asking the user (or we can keep asking the user but apply proper filtering methods).

**Comment by ghammad** [comment]
Although the issue is quite old...
For it is worth, I do confirm that the "resample" option is misleading. I was looking for a way to downsample files that were acquired by mistake at 100Hz instead of 50Hz and I thought I could use the resample option. It seems it is not the case.

Any update on this issue?
Should we move to actipy?

Thank you for your help.

**Comment by chanshing** [comment]
> Although the issue is quite old... For it is worth, I do confirm that the "resample" option is misleading. I was looking for a way to downsample files that were acquired by mistake at 100Hz instead of 50Hz and I thought I could use the resample option. It seems it is not the case.
> 
> Any update on this issue? Should we move to actipy?
> 
> Thank you for your help.

Yes it is a badly named config name that we should rename. Sorry for the confusion. Actipy is one alternative option if resample and lowpass filtering is wanted.
